### PR TITLE
Restructure of folders in output/data

### PR DESCRIPTION
**Pull request by pdeiml**
This PR handles the ideas from #198.

The new structure in the output folder is similar to the one in the input folder. The only difference is the "missing" info.yaml files which are not necessary in the output folder.

@cdeil If you agree with the first commit, I will do a "update output" commit and fix the errors of travis CI which will occur afterwards

### CODE REVIEW

**Comment by cdeil** [comment]
@pdeiml - if you look at the diff here, it's clear that you're basically doing the same change in three places. This is hard to maintain. Can you please call a `make_output_path` function instead and call it three times?

**Comment by pdeiml** [comment]
You are right. I thought about a helper function like this:

def _make_output_path(self, meta, path):
if meta['file_id'] != 1:
path = path / '{}-{}-{}.ecsv'.format(....)
else:
path = path / '{}-{}.ecsv'

but this gets compilcated due to the difference in the file ending for lc, seds and datasets and due to the missing "ds" in the filename of the dataset.

The only possibility is to write this helper function only for the ecsv-files but this is not really a gain.

**Comment by cdeil** [comment]
@pdeiml - I can have a look and try to simplify the code that makes the output path this afternoon.
Can't we remove most of that code and just use the input path, or only modify the input path slightly?

**Comment by pdeiml** [comment]
What do you mean by "that code"?

**Comment by cdeil** [comment]
> What do you mean by "that code"?

The code in `gammacat/collection.py` that generates the path in the output folder.

**Comment by cdeil** [comment]
Turns out I didn't have time to work on this today to improve the code.
Fixed the test. Merging now.

File: gammacat/collection.py [hunk 1]
```diff
@@ -81,16 +81,25 @@ def make_filename(self, meta, *, relative_to_index):
...
 
-        path = path / 'data' / meta['reference_folder']
+        path = path / 'data' / meta['year']
 
-        tag = GammaCatStr.dataset_filename(meta)
+        tag = GammaCatStr.data_filename(meta)
 
         if meta['datatype'] == 'sed':
-            path = path / '{}_sed.ecsv'.format(tag)
+            if meta['file_id'] != 1:
```
Comment (cdeil) [inline gammacat/collection.py]: @pdeiml - Just FYI: there's a typo here, where you have "1" but it should be "-1", leading to wrong stuff:
```
	renamed:    output/data/2016ApJ...817L...7A/gammacat_2016ApJ...817L...7A_000014_lc.ecsv -> output/data/2016/2016ApJ...817L...7A/tev-000014-lc.ecsv
	renamed:    output/data/2016ApJ...817L...7A/gammacat_2016ApJ...817L...7A_000014_sed.ecsv -> output/data/2016/2016ApJ...817L...7A/tev-000014-sed--1.ecsv
	renamed:    output/data/2016ApJ...819..156B/gammacat_2016ApJ...819..156B_000049_lc.ecsv -> output/data/2016/2016ApJ...819..156B/tev-000049-lc-1.ecsv
	new file:   output/data/2016/2016ApJ...819..156B/tev-000049-sed-2.ecsv
	new file:   output/data/2016/2016ApJ...819..156B/tev-000049-sed-3.ecsv
	renamed:    output/data/2016ApJ...819..156B/gammacat_2016ApJ...819..156B_000049_sed.ecsv -> output/data/2016/2016ApJ...819..156B/tev-000049-sed-4.ecsv
	new file:   output/data/2016/2016ApJ...819..156B/tev-000049-sed.ecsv
	renamed:    output/data/2016ApJ...819..156B/gammacat_2016ApJ...819..156B_000049_ds.yaml -> output/data/2016/2016ApJ...819..156B/tev-000049.yaml
	renamed:    output/data/2016ApJ...821..129A/gammacat_2016ApJ...821..129A_000106_sed.ecsv -> output/data/2016/2016ApJ...821..129A/tev-000106-sed--1.ecsv
	renamed:    output/data/2016ApJ...821..129A/gammacat_2016ApJ...821..129A_000106_ds.yaml -> output/data/2016/2016ApJ...821..129A/tev-000106.yaml
	renamed:    output/data/2016ApJ...821..129A/gammacat_2016ApJ...821..129A_000110_sed.ecsv -> output/data/2016/2016ApJ...821..129A/tev-000110-sed--1.ecsv
	renamed:    output/data/2016ApJ...821..129A/gammacat_2016ApJ...821..129A_000110_ds.yaml -> output/data/2016/2016ApJ...821..129A/tev-000110.yaml
	renamed:    output/data/2016ApJ...821..129A/gammacat_2016ApJ...821..129A_000165_ds.yaml -> output/data/2016/2016ApJ...821..129A/tev-000165.yaml
	renamed:    output/data/2016MNRAS.459.2286A/gammacat_2016MNRAS.459.2286A_000043_ds.yaml -> output/data/2016/2016MNRAS.459.2286A/tev-000043.yaml
	renamed:    output/data/2016MNRAS.459.2550A/gammacat_2016MNRAS.459.2550A_000105_lc.ecsv -> output/data/2016/2016MNRAS.459.2550A/tev-000105-lc.ecsv
	renamed:    output/data/2016MNRAS.459.2550A/gammacat_2016MNRAS.459.2550A_000105_sed.ecsv -> output/data/2016/2016MNRAS.459.2550A/tev-000105-sed--1.ecsv
```

**Comment by cdeil** [comment]
Another issue:
```
$ ls -1 input/data/2016/2016ApJ...819..156B
info.yaml
tev-000049-1.yaml
tev-000049-10.yaml
tev-000049-11.yaml
tev-000049-2.yaml
tev-000049-3.yaml
tev-000049-4.yaml
tev-000049-5.yaml
tev-000049-6.yaml
tev-000049-7.yaml
tev-000049-8.yaml
tev-000049-9.yaml
tev-000049-lc-1.ecsv
tev-000049-sed-1.ecsv
tev-000049-sed-2.ecsv
tev-000049-sed-3.ecsv
tev-000049-sed-4.ecsv
tev-000049.yaml
$ ls -1 output/data/2016/2016ApJ...819..156B
tev-000049-lc-1.ecsv
tev-000049-sed-1.ecsv
tev-000049-sed-2.ecsv
tev-000049-sed-3.ecsv
tev-000049-sed-4.ecsv
tev-000049.yaml
```
I think the problem is that those YAML files are missing a file_id, and we don't have checks for that.

I'll try to fix everything in master now, but @pdeiml - if you could have another look tomorrow, that would be great!

**Comment by cdeil** [comment]
I think the output dataset index is also broken ...

**Pull request merged**